runsc/cgroup/cgroup.go

```diff
@@ -542,6 +543,35 @@ func (c *cgroupV1) Install(res *specs.LinuxResources) error {
 	return nil
 }
 
+// Set sets the cgroup resources.
+func (c *cgroupV1) Set(res *specs.LinuxResources) error {
+	log.Debugf("Setting cgroup resources for %q", c.Name)
+	for key, ctrlr := range controllers {
+		if !c.Own[key] {
+			// cgroup is managed by caller, don't touch it.
+			continue
+		}
+		path := c.MakePath(key)
+		if _, err := os.Stat(path); err != nil {
+			if os.IsNotExist(err) && ctrlr.optional() {
+				if err := ctrlr.skip(res); err != nil {
+					return err
+				}
+				log.Infof("Skipping cgroup %q, path doesn't exist", key)
+				continue
+			} else if err != nil {
```

Comment:
`if err != nil` is always true here since it was checked on line 555.

I think the `else` clause is not needed at all since the previous if condition ends with `continue`. You can just return error here (helps save indentation).

Reply:
ah right good catch, i agree.